plugins/destination/mysql/client/client.go

```diff
@@ -52,13 +56,9 @@ func New(ctx context.Context, logger zerolog.Logger, spec specs.Destination) (de
 	db.SetMaxOpenConns(10)
 	db.SetMaxIdleConns(10)
 
-	return &Client{logger: logger.With().Str("module", "mysql").Logger(), db: db, spec: spec, mySQLSpec: mySQLSpec}, nil
+	return c, nil
 }
 
 func (c *Client) Close(ctx context.Context) error {
 	return c.db.Close()
```

Comment:
```suggestion
	if err := c.writer.Close(ctx); err != nil {
		return err
	}
	return c.db.Close()
```

Reply:
we still need to add `_ = c.db.Close()` before `return err`, though